lib/readline.js

```diff
@@ -500,8 +514,49 @@ Interface.prototype._insertString = function(c) {
     // a hack to get the line refreshed if it's needed
     this._moveCursor(0);
   }
+
+  searchHistory.call(this);
 };
 
+function appendSearchResult(result) {
+  this.previewResult = result !== '' ?
+    `\u001b[90m // ${result}\u001b[39m` : '';
+
+  // Cursor to left edge.
+  cursorTo(this.output, 0);
+  clearScreenDown(this.output);
+
+  this.output.write(this.reverseSearchPrompt + this.line + this.previewResult);
+  // now move back the cursor to the original position
+  cursorTo(this.output, this.cursor + this.reverseSearchPrompt.length);
+}
+
+function searchHistory() {
+
+  const searchFor = (array, text) => {
+    const result = [];
+    for (var i = 0; i < array.length; i++)
+      if (text.trim() !== '' && array[i].includes(text))
+        result.push(array[i]);
+    return result;
```

Comment:
As a suggestion: I would not search the whole history each time but only search until the first match and continue searching from that index if the user wants a different entry.